youtube_dl/extractor/nytimes.py

```diff
@@ -139,15 +178,36 @@ class NYTimesArticleIE(NYTimesBaseIE):
             'uploader': 'Matthew Williams',
         }
     }, {
+        'url': 'http://www.nytimes.com/2016/10/14/podcasts/revelations-from-the-final-weeks.html',
+        'md5': 'e0d52040cafb07662acf3c9132db3575',
+        'info_dict': {
+            'id': '20',
+            'title': "The Run-Up: He Was Like an Octopus",
+            'ext': 'mp3',
+            'description': 'We go behind the story of the two women who told us that Donald Trump touched them inappropriately (which he denies) and check in on Hillary Clinton’s campaign.',
+        }
+    }, {
+        'url': 'http://www.nytimes.com/2016/10/16/books/review/inside-the-new-york-times-book-review-the-rise-of-hitler.html',
+        'md5': '66fb5471d7ef15da98af176dc1af4cb9',
+        'info_dict': {
+            'id': 'inside-the-new-york-times-book-review-the-rise-of-hitler',
+            'title': "The Rise of Hitler",
+            'ext': 'mp3',
+            'description': 'Adam Kirsch discusses Volker Ullrich\'s new biography of Hitler; Billy Collins talks about his latest collection of poems; and iO Tillett Wright on his new memoir, "Darling Days."',
+            }
+    }, {
         'url': 'http://www.nytimes.com/news/minute/2014/03/17/times-minute-whats-next-in-crimea/?_php=true&_type=blogs&_php=true&_type=blogs&_r=1',
         'only_matching': True,
     }]
 
     def _real_extract(self, url):
-        video_id = self._match_id(url)
-
-        webpage = self._download_webpage(url, video_id)
+        page_id = self._match_id(url)
 
-        video_id = self._html_search_regex(r'data-videoid="(\d+)"', webpage, 'video id')
+        webpage = self._download_webpage(url, page_id)
 
-        return self._extract_video_from_id(video_id)
+        video_id = self._html_search_regex(r'data-videoid="(\d+)"', webpage, 'video id', None, False)
+        if video_id is not None:
+            return self._extract_video_from_id(video_id)
+        
+        data_json = self._html_search_regex(r'NYTD.FlexTypes.push\(({[^)]*)\)', webpage, 'json data')
```

Comment:
Hopefully this is more to your liking now?

It would have been helpful to have had an example on how this is best used, perhaps a comment in `common.py` where `parse_json` is defined, or somewhere else.
I started looking for examples and first up was `abc.py` which used `[^)]+`:
```
    r'inline(?P<type>Video|Audio|YouTube)Data\.push\((?P<json_data>[^)]+)\);'
```

`adultswim.py` uses `({.*});`
```
    r'var bootstrappedData = ({.*});'
```

It'd be nice to know what the best practice is.
There are a lot of places in the codebase where the same thing is done many different ways and it's hard to know which is preferred…